src/Linters/LinterTrait.hack

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use type Facebook\HHAST\File;
+use namespace HH\Lib\{C, Str};
+
+<<__ConsistentConstruct>>
```

Comment:
if we want more BC in the future, this should be removed, and a static factory method specified instead.

Reply:
I let the constructor be private, therefore no subclass is able to create another constructor and call `parent()`. The signature of the private constructor can be changed while keeping backward compatibility.